src/signed.rs

```diff
@@ -90,7 +91,7 @@ pub struct UncheckedSigned<T: Signable, S> {
 /// to upgrade this `struct` to [`Signed<'a, T, KB>`] which ensures that the signature matches the
 /// signed object, and the method [`UncheckedSigned::check_multi`] can be used to upgrade to
 /// [`Multisigned<'a, T, MK>`].
-impl<T: Signable, S: Signature> UncheckedSigned<T, S> {
```

Comment:
Why did you remove that bound? Is some linter complaining that it technically isn't used?

Reply:
I don't remember why did the compiler complain. In general there is a question whether the  bounds should be in the struct definition or in the `impl` blocks. It seems like the convention in Rust is to put them in the definition, so I restored the bound and propagated it to all `impl` s